executor/slow_query.go

```diff
@@ -465,3 +519,161 @@ func ParseTime(s string) (time.Time, error) {
 	}
 	return t, err
 }
+
+type logFile struct {
+	file       *os.File  // The opened file handle
+	begin, end time.Time // The start/end time of the log file
+}
+
+// File is exported for tests.
+func (l *logFile) File() *os.File {
+	return l.file
+}
+
+// GetAllFiles is used to get all slow-log need to parse, it is export for test.
+func (e *SlowQueryRetriever) GetAllFiles(sctx sessionctx.Context, logFilePath string) ([]logFile, error) {
+	if e.extractor == nil || !e.extractor.Enable {
+		file, err := os.Open(logFilePath)
+		if err != nil {
+			return nil, err
+		}
+		return []logFile{{file: file}}, nil
+	}
+	var logFiles []logFile
+	logDir := filepath.Dir(logFilePath)
+	ext := filepath.Ext(logFilePath)
+	prefix := logFilePath[:len(logFilePath)-len(ext)]
+	handleErr := func(err error) error {
+		// Ignore the error and append warning for usability.
+		if err != io.EOF {
+			sctx.GetSessionVars().StmtCtx.AppendWarning(err)
+		}
+		return nil
+	}
+	err := filepath.Walk(logDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return handleErr(err)
+		}
+		if info.IsDir() {
+			return nil
+		}
+		//All rotated log files have the same prefix with the original file
+		if !strings.HasPrefix(path, prefix) {
+			return nil
+		}
+		file, err := os.OpenFile(path, os.O_RDONLY, os.ModePerm)
+		if err != nil {
+			return handleErr(err)
+		}
+		skip := false
+		defer func() {
+			if !skip {
+				terror.Log(file.Close())
+			}
+		}()
+		// Get the file start time.
+		fileBeginTime, err := e.getFileStartTime(file)
+		if err != nil {
+			return handleErr(err)
+		}
+		if fileBeginTime.After(e.extractor.EndTime) {
+			return nil
+		}
+
+		// Get the file end time.
+		fileEndTime, err := e.getFileEndTime(file)
+		if err != nil {
+			return handleErr(err)
+		}
+		if fileEndTime.Before(e.extractor.StartTime) {
+			return nil
+		}
+		_, err = file.Seek(0, io.SeekStart)
+		if err != nil {
+			return handleErr(err)
+		}
+		logFiles = append(logFiles, logFile{
+			file:  file,
+			begin: fileBeginTime,
+			end:   fileEndTime,
+		})
+		skip = true
+		return nil
+	})
+	// Sort by start time
+	sort.Slice(logFiles, func(i, j int) bool {
+		return logFiles[i].begin.Before(logFiles[j].begin)
+	})
+	return logFiles, err
+}
+
+func (e *SlowQueryRetriever) getFileStartTime(file *os.File) (time.Time, error) {
+	var t time.Time
+	_, err := file.Seek(0, io.SeekStart)
+	if err != nil {
+		return t, err
+	}
+	reader := bufio.NewReader(file)
+	maxNum := 128
+	for {
+		lineByte, err := getOneLine(reader)
+		if err != nil {
+			return t, err
+		}
+		line := string(lineByte)
+		if strings.HasPrefix(line, variable.SlowLogStartPrefixStr) {
+			return ParseTime(line[len(variable.SlowLogStartPrefixStr):])
+		}
+		maxNum -= 1
+		if maxNum <= 0 {
+			break
+		}
+	}
+	return t, errors.Errorf("can't found start time")
```

Comment:
```suggestion
	return t, errors.Errorf("malform slow query file %v", file.Name())
```
It's better to attach the file name information.